components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4908,6 +4914,77 @@ public String getExternalWorkflowReferenceId(int subscriptionId) throws APIManag
         return apiMgtDAO.getExternalWorkflowReferenceForSubscription(subscriptionId);
     }
 
+    @Override
+    public int addCertificate(String userName, String certificate, String alias, String endpoint)
+            throws APIManagementException {
+        ResponseCode responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+        ;
+        try {
+            int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(tenantDomain);
+            responseCode = certificateManager
```

Comment:
add certificate to publisher. Whay we need to include publsiher in method signature?

Reply:
This is a common class which will be called both publisher and gateway nodes. 
i.e when adding the certificate to the publisher trust store, it will call addCertificatetoPublisher. When adding to the gateway, it will call addCertificateToGateway in the gateway node. 
I used publisher and gateway to distinguish publisher and gateway method calls.